### PR TITLE
Fixed Housing Search API Transaction Search Shared Model

### PR DESCRIPTION
**Pull request by sandipchandra**
## Link to JIRA ticket

Fixed Housing Search API Transaction Search Shared Model

## Describe this PR

Fixed Housing Search API Transaction Search Shared Model

Add a summary of the problem and any additional context that helps explain what the issue we're trying to solve is.

### *What changes have we introduced*

Fixed Housing Search API Transaction Search Shared Model with changing models properties to Text type in QueryableTransaction model.

Include any links to commits that introduce significant additions of code if they help explain the process of coming to the solution. e.g Addition of test database setup, addition of first test and production class, removal of buggy code, etc.

#### _Checklist_

- [true ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [true  ] Added tests to cover all new production code
- [true  ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ true ] Checked all code for possible refactoring
- [ true ] Code pipeline builds correctly

### *Follow up actions after merging PR*

Are there any next steps that need to addressed after merging this PR? Add them here.

### CODE REVIEW

File: Hackney.Shared.HousingSearch/Domain/Transactions/Sender.cs [hunk 1]
```diff
@@ -4,6 +4,10 @@ namespace Hackney.Shared.HousingSearch.Domain.Transactions
 {
     public class Sender
     {
+        public Sender()
```
Comment (AnnaGolosova) [inline Hackney.Shared.HousingSearch/Domain/Transactions/Sender.cs]: Why do we need empty constuctors here?

Comment (AnnaGolosova) [reply Hackney.Shared.HousingSearch/Domain/Transactions/Sender.cs]: I think the main idea here is to have read-only domain models and to transfer responsibility of object creation to the `Create` method. This is requirenments from @amidomichael

File: Hackney.Shared.HousingSearch/Domain/Transactions/SuspenseResolutionInfo.cs [hunk 2]
```diff
@@ -4,6 +4,10 @@ namespace Hackney.Shared.HousingSearch.Domain.Transactions
 {
     public class SuspenseResolutionInfo
     {
+        public SuspenseResolutionInfo()
```
Comment (amidomichael) [inline Hackney.Shared.HousingSearch/Domain/Transactions/SuspenseResolutionInfo.cs]: can this be removed?

File: Hackney.Shared.HousingSearch/Domain/Transactions/SuspenseResolutionInfo.cs [hunk 3]
```diff
@@ -17,9 +21,9 @@ private SuspenseResolutionInfo(DateTime? resolutionDate, bool isConfirmed, bool
 
         public bool IsResolve => IsConfirmed && IsApproved;
 
-        public DateTime? ResolutionDate { get; }
-        public bool IsConfirmed { get; }
-        public bool IsApproved { get; }
-        public string Note { get; }
+        public DateTime? ResolutionDate { get; set; }
```
Comment (amidomichael) [inline Hackney.Shared.HousingSearch/Domain/Transactions/SuspenseResolutionInfo.cs]: can these setters be private?

File: Hackney.Shared.HousingSearch/Domain/Transactions/Sender.cs [hunk 4]
```diff
@@ -13,8 +17,8 @@ private Sender(Guid id, string fullName)
             FullName = fullName;
         }
 
-        public Guid Id { get; }
+        public Guid Id { get; set; }
```
Comment (amidomichael) [inline Hackney.Shared.HousingSearch/Domain/Transactions/Sender.cs]: can these setters be private?

File: Hackney.Shared.HousingSearch/Domain/Transactions/Transaction.cs [hunk 5]
```diff
@@ -6,6 +6,10 @@ namespace Hackney.Shared.HousingSearch.Domain.Transactions
 {
     public class Transaction
     {
+        public Transaction()
```
Comment (amidomichael) [inline Hackney.Shared.HousingSearch/Domain/Transactions/Transaction.cs]: remove?

File: Hackney.Shared.HousingSearch/Domain/Transactions/Transaction.cs [hunk 6]
```diff
@@ -48,75 +52,75 @@ public class Transaction
             Fund = fund;
         }
 
-        public Guid Id { get; }
+        public Guid Id { get; set; }
```
Comment (amidomichael) [inline Hackney.Shared.HousingSearch/Domain/Transactions/Transaction.cs]: can all of the setters be private?

**Review by AnnaGolosova — APPROVED** [review]

**Review by amidomichael — APPROVED** [review]

**Pull request merged**